cerbero/packages/osx/packager.py

```diff
@@ -302,59 +311,70 @@ def _create_product(self, package_type):
         self.package.set_mode(package_type)
         m.action(_("Creating Distribution.xml for package %s " % self.package))
         distro = DistributionXML(self.package, self.store, self.tmp,
-            self.packages_paths[package_type],
-            self.empty_packages[package_type], package_type,
-            self.config.target_arch, home_folder=self.home_folder)
+                                 self.packages_paths[package_type],
+                                 self.empty_packages[package_type], package_type,
+                                 self.config.target_arch, home_folder=self.home_folder)
         distro_path = os.path.join(self.tmp, "Distribution.xml")
         distro.write(distro_path)
         output_file = os.path.join(self.output_dir, self._package_name('.pkg'))
         output_file = os.path.abspath(output_file)
         pb = ProductBuild()
-        pb.create_package(distro_path, output_file, [self.package.relative_path('.')])
+        search_paths = [self.package.relative_path('.')]
+        search_paths += set([os.path.dirname(p) for p in self.packages_paths[package_type].values()])
+        pb.create_package(distro_path, output_file, search_paths)
         return output_file
 
     def _create_packages(self):
+        extra_paths = []
         for p in self.packages:
             m.action(_("Creating package %s ") % p)
-            packager = OSXPackage(self.config, p, self.store)
+            p.pre_package()
+            packager = OSXPackager(self.config, p, self.store)
+            paths = [None, None]
             try:
-                paths = packager.pack(self.output_dir, self.devel, self.force,
+                paths = packager.pack(self.tmp, self.devel, self.force,
                                       self.keep_temp, self.split, self.package.version,
                                       install_dir=self.package.get_install_dir(),
                                       include_dirs=self.include_dirs,
                                       sdk_version=self.package.sdk_version)
                 m.action(_("Package created sucessfully"))
             except EmptyPackageError:
-                paths = [None, None]
+                pass
+            finally:
+                paths = p.post_package(paths, self.output_dir)
 
             if paths[0] is not None:
                 self.packages_paths[PackageType.RUNTIME][p] = paths[0]
             else:
                 self.empty_packages[PackageType.RUNTIME].append(p)
+
             if paths[1] is not None:
                 self.packages_paths[PackageType.DEVEL][p] = paths[1]
             else:
                 self.empty_packages[PackageType.DEVEL].append(p)
 
-    def _create_packages_dmg(self):
-        paths = list(self.packages_paths[PackageType.RUNTIME].values())
+            extra_paths += paths[2:]
```

Comment:
Can't you use the same dictionary self.packages_paths used to accumulate paths ?

Reply:
no because this dictionary is used at different places expecting only one path and nothing more.
This mechanism expects that all packages only produce 2 outputs: runtime and devel.
In fluendo we may produce more than 2 outputs (like the nuget packages in fluendo-sdk) and I found the idea interesting to be able to archive those extra packages too in the dmg archive in order to distribute them with the sdk installer package itself.